src/main/java/mSearch/filmlisten/FilmlisteLesen.java

```diff
@@ -119,125 +82,187 @@ private InputStream selectDecompressor(String source, InputStream in) throws Exc
         return in;
     }
 
-    public void readFilmListe(String source, final ListeFilme listeFilme, int days) {
-        Log.sysLog("Liste Filme lesen von: " + source);
+    private void readData(JsonParser jp, ListeFilme listeFilme) throws IOException {
         JsonToken jsonToken;
         String sender = "", thema = "";
-        listeFilme.clear();
-        this.notifyStart(source, PROGRESS_MAX); // für die Progressanzeige
 
-        if (days > 0) {
-            final long maxDays = 1000L * 60L * 60L * 24L * days;
-            seconds = new Date().getTime() - maxDays;
-        } else {
-            seconds = 0;
+        if (jp.nextToken() != JsonToken.START_OBJECT) {
+            throw new IllegalStateException("Expected data to start with an Object");
         }
 
-        try (InputStream in = selectDecompressor(source, getInputStreamForLocation(source));
-                JsonParser jp = new JsonFactory().createParser(in)) {
-
-            if (jp.nextToken() != JsonToken.START_OBJECT) {
-                throw new IllegalStateException("Expected data to start with an Object");
+        while ((jsonToken = jp.nextToken()) != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS135)